### PR TITLE
Refactor ReimbursementAccountPage's fetchFreePlanVerifiedBankAccount

### PR DESCRIPTION
**Pull request by ctkochan22**
@nkuoch @MariaHCD @youssef-lr 

### Details
This replaces the use of fetchFreePlanVerifiedBankAccount 

### Fixed Issues
$  https://github.com/Expensify/Expensify/issues/226858

### Tests / QA

**Test a bank account in SETUP**
1. Start the VBBA process. Select plaid (use `user_good` and `pass_good` to get past plaid)
2. Add any fields to the Company Step, the key is to make it past this step
3. After submission, you should be on the RequestorStep "Personal Information"
    image
4. Close out of the flow. Back all the way to a chat window
5. Navigate back to the VBA flow (Settings -> Workspace settings -> Connect Bank Account -> Continue with setup)
6. Verify that you are shown back to the Requestor Step "Personal information"
7. Now log out to clear your cache
8. Log back in
9. Navigate back to the VBA flow (Settings -> Workspace settings -> Connect Bank Account)
10. Verify that you are shown back to the Requestor Step "Personal information"

**Test Requestor Step without completing Onfido**
1. With the same VBBA setup, fill out the requestor step (_Do not use Alberta Charleston from the SO, use a random name_)
2. It should take you to Onfido flow
3. Don't complete onfido. Exit the flow
4. Navigate back to the VBA flow (Settings -> Workspace settings -> Connect Bank Account -> Continue with setup)
5. Verify that you are brought back to the Onfido Step
6. Log out and log back in
7. Navigate back to the VBA flow
8. Verify that you are brought back to the Requestor Step ("Personal information"). And everything is pre-filled
    image
9. When you click submit, you still have to go through Onfido

**Test Beneficial Owners Step**
1. Once onfido is complete, you should be moved to the ACHContract Step or Beneficial Owners Step or "Additional information"
    image
2. Exit flow
3. Navigate back to the VBA flow (Settings -> Workspace settings -> Connect Bank Account -> Continue with setup)
5. Verify that you are still on that step you left off
6. Log out and log back in
7. Navigate back to the VBA flow
8. Verify that you are still on that step

**Test VERIFYING state**
1. So long as you did not add Alberta Charleston as the name, the VBBA should be in a VERIFYING state
2. Verify that you are asked to provide more information:
    image
3. Click "Yes lets chat"
4. Navigate back to the VBA flow (Settings -> Workspace settings -> Connect Bank Account -> Continue with setup)
5. Verify that you are still on that step/need to provide more info
6. Log out and log back in.
7. Navigate back to the VBA flow 
8. Verify that you are still on that step

**Test PENDING state**
1. Follow the SO instructions to add a PENDING VBBA ([link](https://stackoverflow.com/c/expensify/questions/342)). 
2. Get to the Validation Step
    image
3. Navigate away
4. Navigate back to the VBBA flow
5. Verify that you see the "Continue Setup"/"Start-Over". 
6. Click continue
7. Verify that you see Validation Step "Test Transactions"

- [x] Verify that no errors appear in the JS console

#### PR Author Checklist
- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

<details>
<summary><h4>PR Reviewer Checklist</h4>

The reviewer will copy/paste it into a new comment and complete it after the author checklist is completed
</summary>

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [ ] MacOS / Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

</details>

#### Desktop
https://recordit.co/fseVdmCoMz

#### iOS
https://recordit.co/ygknviefmP

### CODE REVIEW

File: src/libs/actions/BankAccounts.js [hunk 1]
```diff
@@ -243,6 +243,47 @@ function validateBankAccount(bankAccountID, validateCode) {
...
 }
 
+function openReimbursementAccountPage(stepToOpen, subStep, localCurrentStep) {
+    const onyxData = {
+        optimisticData: [
+            {
+                onyxMethod: CONST.ONYX.METHOD.MERGE,
+                key: ONYXKEYS.REIMBURSEMENT_ACCOUNT,
+                value: {
+                    error: '',
```
Comment (youssef-lr) [inline src/libs/actions/BankAccounts.js:253]: I think we've switched to using `errors` object keyed by microtime?

File: src/libs/actions/BankAccounts.js [hunk 2]
```diff
@@ -243,6 +243,47 @@ function validateBankAccount(bankAccountID, validateCode) {
...
+                    loading: true,
```
Comment (youssef-lr) [inline src/libs/actions/BankAccounts.js:254]: Same goes here I think, `isLoading` instead of `loading`.

**Comment by ctkochan22** [comment]
Still need to figure out QA and screenshots from every platform, but this can begin initial review I think and testing

File: src/pages/ReimbursementAccount/ReimbursementAccountPage.js [hunk 3]
```diff
@@ -157,9 +158,9 @@ class ReimbursementAccountPage extends React.Component {
         // We can specify a step to navigate to by using route params when the component mounts.
         // We want to use the same stepToOpen variable when the network state changes because we can be redirected to a different step when the account refreshes.
         const stepToOpen = this.getStepToOpenFromRouteParams();
-
-        // If we are trying to navigate to `/bank-account/new` and we already have a bank account then don't allow returning to `/new`
-        BankAccounts.fetchFreePlanVerifiedBankAccount(stepToOpen !== CONST.BANK_ACCOUNT.STEP.BANK_ACCOUNT ? stepToOpen : '');
+        const subStep = lodashGet(store.getReimbursementAccountInSetup(), 'subStep', '');
```
Comment (nkuoch) [inline src/pages/ReimbursementAccount/ReimbursementAccountPage.js:161]: extract store.getReimbursementAccountInSetup() in a const

**Comment by youssef-lr** [comment]
I don't know what's up with my VM, I'm getting a CORS error when trying to open the VBBA page. Trying to figure out what's causing this so I can test the PR.

**Review by MariaHCD — COMMENTED** [review]
The code LGTM so far!

I believe we also need to refactor the usage of fetchFreePlanVerifiedBankAccount in the WorkspacePageWithSections component: 
https://github.com/Expensify/App/blob/3a5d99b52fcfef884d75bec8f733d18e1b87f3e3/src/pages/workspace/WorkspacePageWithSections.js#L83-L86

**Comment by MariaHCD** [comment]
> I don't know what's up with my VM, I'm getting a CORS error when trying to open the VBBA page. Trying to figure out what's causing this so I can test the PR.

@youssef-lr Possibly you need to upgrade to PHP8.1?

**Comment by youssef-lr** [comment]
@MariaHCD I've already did. I think I'll switch to staging to test this while I figure it out.

**Review by MariaHCD — COMMENTED** [review]
Tested this locally:
1. Test a bank account in SETUP: worked as expected
2. Test Requestor Step without completing Onfido: worked as expected, except for logging out and logging back in. The personal information form did not have the form data pre-filled

[New Expensify (2).webm](https://user-images.githubusercontent.com/12268372/197195030-481ac733-42ac-4103-a30b-780962b61be5.webm)

3. Test Beneficial Owners Step: worked as expected
4. Test PENDING state: haven't tested this yet. We can remove `At the moment, the SO instructions to add a PENDING VBBA is broken` since that bug has been fixed.

**Review by nkuoch — DISMISSED** [review]

**Comment by youssef-lr** [comment]
OK I figured it out, I needed to checkout Chris's Auth and Web-E branches to test this /facepalm. Testing right now.

**Comment by youssef-lr** [comment]
@ctkochan22 these steps didn't work for me as expected, logging out then back in takes me back to the first step.

- Now log out to clear your cache
- Log back in
- Navigate back to the VBA flow (Settings -> Workspace settings -> Connect Bank Account)
- Verify that you are shown back to the Requestor Step "Personal information"

For some reason uploading a screen recording is not working for me here in GH, I will keep trying.

**Review by MariaHCD — CHANGES_REQUESTED** [review]
This looks good, I think we just need to iron out the tests since they aren't working 100% correctly. I'll retest tomorrow.

In addition, do we need to refactor the usage of fetchFreePlanVerifiedBankAccount in the WorkspacePageWithSections page? 

https://github.com/Expensify/App/blob/3a5d99b52fcfef884d75bec8f733d18e1b87f3e3/src/pages/workspace/WorkspacePageWithSections.js#L83-L86

**Comment by JmillsExpensify** [comment]
What's left to get this PR merged?

**Comment by ctkochan22** [comment]
This should be ready to go @MariaHCD @nkuoch 

The only issue to note is, as noted in this bug https://github.com/Expensify/Expensify/issues/238317 . When testing and you sign in for the first time, there will be no VBBA and we will not fetch it at any point until we click "Continue Setup". This means, we won't see the option to continue previous setup or start over

File: src/libs/actions/ReimbursementAccount/setupWithdrawalAccount.js [hunk 4]
```diff
@@ -58,7 +58,7 @@ function getNextStep(updatedACHData) {
         return currentStep;
     }
 
-    if (currentStep === CONST.BANK_ACCOUNT.STEP.VALIDATION && updatedACHData.bankAccountInReview) {
```
Comment (nkuoch) [inline src/libs/actions/ReimbursementAccount/setupWithdrawalAccount.js:61]: Did you make sure no other places used `bankAccountInReview`?

Comment (ctkochan22) [reply src/libs/actions/ReimbursementAccount/setupWithdrawalAccount.js:61]: Oh I need to undo this. We actually removed `bankAccountInReview` completely from the web/auth side. 

And looking at this page, I don't think this method is used anymore? Its used in `setupWithdrawalAccount()`, but that is no longer being used I don't think.

Comment (ctkochan22) [reply src/libs/actions/ReimbursementAccount/setupWithdrawalAccount.js:61]: I'm going to undo this change

Comment (MariaHCD) [reply src/libs/actions/ReimbursementAccount/setupWithdrawalAccount.js]: setupWithdrawalAccount() isn't being used anymore, created an issue to remove this file completely: https://github.com/Expensify/Expensify/issues/240702

**Review by nkuoch — APPROVED** [review]

**Review by MariaHCD — APPROVED** [review]

**Review by MariaHCD — CHANGES_REQUESTED** [review]
Could you also update the Jest unit tests? We're currently using fetchFreePlanVerifiedBankAccount in [ReimbursementAccountTest.js](https://github.com/Expensify/App/blob/bb2aa73cefd9aee3873efb4480162c7ed2064bee/tests/actions/ReimbursementAccountTest.js)

Can we also get rid of `tests/unit/fetchFreePlanVerifiedBankAccountTest.js`?

**Comment by ctkochan22** [comment]
> Could you also update the Jest unit tests? We're currently using fetchFreePlanVerifiedBankAccount in [ReimbursementAccountTest.js](https://github.com/Expensify/App/blob/bb2aa73cefd9aee3873efb4480162c7ed2064bee/tests/actions/ReimbursementAccountTest.js)

I'm not super familiar with Jest tests. But it seems like it doesn't actually hit an API? If that is the case, is there much to test? We handle so much of it on the php side.

**Comment by MariaHCD** [comment]
> I'm not super familiar with Jest tests. But it seems like it doesn't actually hit an API? If that is the case, is there much to test? We handle so much of it on the php side.

Ah, good point. We're just mocking the API responses in the Jest tests. And since most of the logic is now in the API, we shouldn't need these tests anymore, correct?

**Review by MariaHCD — APPROVED** [review]

**Comment by MariaHCD** [comment]
@ctkochan22 Could you re-fill the PR author checklist? Perhaps it's outdated which is why the test is failing.

**Comment by MariaHCD** [comment]
Noticed when testing this that if you log out and then back in again, the forms that you completed before are no longer populated with the data:


https://user-images.githubusercontent.com/12268372/200581796-97522ea1-58c6-4f96-bcc9-23bdb72c45e3.mov

**Comment by ctkochan22** [comment]
Woah, this is happening because the path here is incorrect.

https://github.com/Expensify/App/blob/4d20ee5ccb7d8772e00cfac9e691a7a0df62d551/src/libs/ReimbursementAccountUtils.js#L26

Was that changed recently?

**Comment by ctkochan22** [comment]
@youssef-lr Can you test this today please? 🙇

**Comment by youssef-lr** [comment]
@ckscanlon testing now!

**Comment by MariaHCD** [comment]
Heads up @youssef-lr, if you run into this error: https://github.com/Expensify/App/pull/11751#issuecomment-1307252840

You can test with the Web-E PR: https://github.com/Expensify/Web-Expensify/pull/35506 and it will work

**Comment by youssef-lr** [comment]
@ckscanlon after running the query in the last step, this is what I'm seeing, am I missing something? All other tests worked as expected.
Screen Shot 2022-11-09 at 15 06 28

**Comment by ctkochan22** [comment]
@youssef-lr  Is the bank account ending in `9378` the new bank account in the validate state?

**Comment by ctkochan22** [comment]
@youssef-lr my bad, the test steps were outdated. Please try the last step when you can! And this PR should be ready to go!

**Comment by ctkochan22** [comment]
I'm not sure why this PR didn't have a C+ assigned

**Review by youssef-lr — APPROVED** [review]
Tested well!

**Comment by youssef-lr** [comment]
<summary><h4>PR Reviewer Checklist</h4>

</summary>

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

File: src/libs/actions/BankAccounts.js [hunk 5]
```diff
@@ -242,6 +242,47 @@ function validateBankAccount(bankAccountID, validateCode) {
...
 
+function openReimbursementAccountPage(stepToOpen, subStep, localCurrentStep) {
+    const onyxData = {
+        optimisticData: [
+            {
+                onyxMethod: CONST.ONYX.METHOD.MERGE,
+                key: ONYXKEYS.REIMBURSEMENT_ACCOUNT,
+                value: {
+                    errors: null,
+                    isLoading: true,
```
Comment (Luke9389) [inline src/libs/actions/BankAccounts.js:253]: is it this @ctkochan22 ?

Are we stuck in the isLoading state?

Comment (ctkochan22) [reply src/libs/actions/BankAccounts.js:253]: but upon any sort of completion it should be set to false

**Pull request merged**